src/main/java/uk/gov/hmcts/reform/prl/courtnav/mappers/FL401ApplicationMapper.java

```diff
@@ -217,17 +227,57 @@ public CaseData mapCourtNavData(CourtNavFl401 courtNavCaseData) throws NotFoundE
             .caseSubmittedTimeStamp(DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(zonedDateTime))
             .build();
 
-        caseData = caseData.toBuilder()
-            .courtName(getCourtName(caseData))
-            .courtEmailAddress(getCourtEmailAddress(court))
-            .build();
-
+        caseData = populateCourtDetailsForCourtNaveCase(authorization, caseData,
+                                                        courtNavCaseData.getMetaData().getCourtSpecialRequirements());
         caseData = caseData.setDateSubmittedDate();
 
         return caseData;
 
     }
 
+    private CaseData populateCourtDetailsForCourtNaveCase(String authorization, CaseData caseData,
+                                                          String epmsId) throws NotFoundException {
```

Comment:
its called EPIMS id please.

Reply:
Do you wanna run a null check on metadata before calling this method?

Reply:
I think null check is not required here as metadata is already getting used before calling this method.